src/main/java/de/chojo/repbot/dao/provider/Guilds.java

```diff
@@ -3,31 +3,75 @@
 import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
 import de.chojo.repbot.dao.access.guild.RepGuild;
+import de.chojo.repbot.dao.access.guild.RepGuildId;
+import de.chojo.repbot.dao.pagination.GuildList;
+import de.chojo.sadu.base.QueryFactory;
 import net.dv8tion.jda.api.entities.Guild;
 import org.slf4j.Logger;
 
 import javax.sql.DataSource;
+import java.util.List;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 
 import static org.slf4j.LoggerFactory.getLogger;
 
-public class Guilds {
+public class Guilds extends QueryFactory {
     private static final Logger log = getLogger(Guilds.class);
-    private final DataSource dataSource;
     private final Cache<Long, RepGuild> guilds = CacheBuilder.newBuilder().expireAfterAccess(30, TimeUnit.MINUTES)
                                                              .build();
 
     public Guilds(DataSource dataSource) {
-        this.dataSource = dataSource;
+        super(dataSource);
     }
 
     public RepGuild guild(Guild guild) {
         try {
-            return guilds.get(guild.getIdLong(), () -> new RepGuild(dataSource, guild)).refresh(guild);
+            return guilds.get(guild.getIdLong(), () -> new RepGuild(source(), guild)).refresh(guild);
         } catch (ExecutionException e) {
             log.error("Could not create guild adapter", e);
             throw new RuntimeException("", e);
         }
     }
+
+    /**
+     * Gets a guild by id. This guild object might have limited functionality. This object is never cached.
+     * It should never be used to change settings.
+     *
+     * @param id
+     * @return
+     */
+    public RepGuild guild(long id) {
+        return new RepGuildId(source(), id);
+    }
+
+    public GuildList guilds(int pageSize) {
+        return new GuildList(() -> pages(pageSize), page -> page(pageSize, page));
+    }
```

Comment:
Added a method to create a guild based on an id instead of a guild object.

Added a method to read all known guilds